### PR TITLE
Juanmaturino/edu 23 need to add example of summarizer component for

### PR DESCRIPTION
**Pull request by jjmaturino**
adding prediction guard to summarizer component to repo

## Description

Addition of [Prediction Guard](https://predictionguard.com/) summarization (DocSum) component to opea-project/GenAIComps repo

## Issues

**N/A**

## Type of change

List the type of change like below. Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds new functionality)

## Dependencies

**N/A**
 
## Tests

Ran the unit tests under tests/llms/...

### CODE REVIEW

**Comment by jjmaturino** [comment]
@sharanshirodkar7 Tagging for observability.

**Comment by jjmaturino** [comment]
@lvliang-intel RFR 🛸

**Comment by jjmaturino** [comment]
Closing PR Due To Rebasing issues.

**Comment by jjmaturino** [comment]
Closing PR Due To Rebasing issues. Will Reopen.

**Comment by jjmaturino** [comment]
Reopened PR, after fixing git commit merging errors :) 🚀 🚀 🚀

**Comment by jjmaturino** [comment]
@lianhao can you please review :)

File: comps/llms/summarization/predictionguard/llm_predictionguard.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    port=9000,
+)
+@register_statistics(names=["opea_service@llm_predictionguard_docsum"])
+def llm_generate(input: LLMParamsDoc):
+    start = time.time()
+
+    messages = [
+        {
+            "role": "system",
+            "content": "You are a summarization assistant. Your goal is to provide a very concise, summarized responses of the user query.",
```
Comment (ftian1) [inline comps/llms/summarization/predictionguard/llm_predictionguard.py:38]: @lvliang-intel pls review this PR. 

From my view, why not merge this with guardrail component we have had? looks like only the prompt and model variant are different

Comment (jjmaturino) [reply comps/llms/summarization/predictionguard/llm_predictionguard.py:38]: @ftian1 Hey Tian,

Hope you're having a good day.

> **TL;DR:** I want to add a document summarization component (PredictionGuard DocSum) to their GenAI Example Repo. This will also allow flexibility in how doc summaries are generated without altering the existing text generation component.


Adding this summarization component, is desired, because it relies on the [**PredictionGuard** ](https://docs.predictionguard.com/api-reference/api-reference/completions) platform.

I want to add a PredictionGuard DocSum Permutation to the  GenAI Example Repo at some point, and the document summarization component is a necessary component for that _MegaService_.


You are correct, that this component only differs in the prompt and model from the predictionguard/text-gen component.

Having two separate components allows us to separate concerns if in the future we want to change how docsum is handled by predictionguard, without modifying textgen

**Comment by xiguiw** [comment]
@jjmaturino 

The micros-service is duplicated to DocSum in function.
Is this still active?

**Comment by jjmaturino** [comment]
@xiguiw  I need to review the changes made to the codebase. I saw that great improvements were made to the code architecture. I need to see which of my previous work is irrelevant now, and update to make the required changes.

**Comment by jjmaturino** [comment]
@xiguiw Okay I just reviewed the changes. Yes, this was essentially an _integration_ for the new doc-summarization directory. Will refactor this pr to adhere to new structure. 

Thanks for you help.

**Comment by jjmaturino** [comment]
Closing PR due to significant changes from start and current state of main. Will open new pr with desired changes.
